source/agora/consensus/EnrollmentManager.d

```diff
@@ -703,6 +703,23 @@ public class EnrollmentManager
         return true;
     }
 
+    /***************************************************************************
+
+        Get the pre-image hash for this validator
+
+        Params:
+            height = the block height of pre-image to be returned
+
+        Returns:
+            The preimage hash if found otherwise Hash.init
+
+    ***************************************************************************/
+
+    public Hash getOurPreimage (in Height height) @safe nothrow
+    {
+        return this.cycle[height];
+    }
```

Comment:
https://github.com/bosagora/agora/pull/2113

Oh, you've already created this function.
I need this :)

Reply:
Hopefully merged soon.

Reply:
@TrustHenry: I also split this into a separate commit if you want to cherry-pick it before it is merged.